channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -608,9 +631,16 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
                     _self._buffer.markAsSent(payload);
                     // no response from beaconSender, clear buffer
                     _self._onSuccess(payload, payload.length);
+                    if (_self._statsbeat && _self._statsbeat.isInitialized()) {
+                        let endTime = dateNow();
+                        _self._statsbeat.countRequest(url, endTime - startTime, true);
+                    }
                 } else {
-                    _xhrSender(payload, true);
+                    _xhrSender(payload, true, startTime);
                     _self.diagLog().throwInternal(LoggingSeverity.WARNING, _InternalMessageId.TransmissionFailed, ". " + "Failed to send telemetry with Beacon API, retried with xhrSender.");
+                    if (_self._statsbeat && _self._statsbeat.isInitialized()) {
```

Comment:
Failure to send with beacon api counts as exception? What about the result of xhrSender? Do we count that as well?

Reply:
Failed to send with any sender counts as exception. For other senders, they call `_onError` inside which is calling statsbeat.countException. Are you saying if sendbeacon fails then xhrsender also fails. In that case we'll double counting exceptions? I'm not sure if this count as duplicate - because they're both exceptions isn't it? @MSNev what do you think?

Reply:
There will be no double counting, the Sender only uses 1 transport, beacon, xhr, xdr or fetch.